examples/server/server.cpp

```diff
@@ -1940,6 +1946,12 @@ struct server_context {
                             send_final_response(slot);
                             continue;
                         }
+                        // context shift is disabled and prompt is too large - discard it
+                        if (!params.ctx_shift && (slot.n_prompt_tokens > slot.n_ctx) ){
```

Comment:
I don't know if this leads to correct behavior. I found that if we do:
```
slot.n_prompt_tokens > slot.n_ctx
```
Then, it's possible to fall through the check down to prompt truncation which might be confusing for the user. Maybe we should change it to:
```
slot.n_prompt_tokens >= slot.n_ctx
```
Maybe someone more knowledgeable could chime in.